tests/integration/telemetry/api/registry_setup_test.go

```diff
@@ -34,11 +34,25 @@ const (
 )
 
 func testRegistrySetup(ctx resource.Context) (err error) {
-	registry, err = registryredirector.New(ctx, registryredirector.Config{
-		Cluster:        ctx.AllClusters().Default(),
-		TargetRegistry: "kind-registry:5000",
-		Scheme:         "http",
-	})
+	var config registryredirector.Config
+
+	isKind, err := IsKindCluster()
+	if err != nil {
+		return
+	}
+
+	// By default, for any platform, the test will pull the test image from public "gcr.io" registry.
+	// For "Kind" environment, it will pull the images from the "kind-registry".
+	// For "Kind", this is due to DNS issues in IPv6 cluster
+	if isKind {
+		config = registryredirector.Config{
```

Comment:
Nice!

A theoretical future addition could be to also check the registry has the image. Since right now, if I run on kind ipv4 without running the `crane copy ..` setup it would fail.

But this is like niche^2 use case and probably not worth the complexity

Reply:
Thank you.

I'll add this as a future task to my ToDo list.